### ISSUE TITLE
Strange behaviour of using ctx and integer arrays

### ISSUE THREAD

**Issue by lingyiyang** [datasig-ac-uk/RoughPy#70]
There seems to be unexpected behaviour when using integer arrays. Namely we observe that for increments of length two, if the data is passed in as integers, then it inteprets the paths as 2D no matter what argument is passed for the width. This changes when using floats. However, this unexpected behaviour still persist even in the case of floats if we pass parameters in via ctx argument (for increments up to length two).

A minor comment seems to be that there is a difference in passing coeffs in through ctx or just as an argument in the constructor of the path.

I have created a notebook of examples illustrating these here: https://github.com/lingyiyang/RoughPy/blob/main/tests/array_behaviour.ipynb

Finally, if a one-dimensional array is passed in `LieIncrementStream.from_increments`, then the kernel automatically dies. If this function cannot handle such data, I would not expect the entire kernel to crash.

**Comment by inakleinbottle** [comment]
Very quickly: the crashing kernel is likely due to a segmentation fault. Typically this is happening because the Python interpreter gets a null value that it is not expecting, which usually means I have missed setting the Python error state before returning null. The other issues are more complicated to track down, but I suspect they are due to the conversion that takes place because the values are not of the correct type going in. I will investigate.

**Comment by lingyiyang** [comment]
v0.1.0

**Comment by inakleinbottle** [comment]
@lingyiyang Ok, so I think some of these issues have been resolved in 0.1.1 (the current release version). Can you update RoughPy and confirm this?

I did notice a problem in the following step:
```python
stream = rp.LieIncrementStream.from_increments(np.array([[3, 7, 0, 4], [0, 0, 1, 5]]), width=2, depth=2, coeffs=rp.Rational) 
sig = stream.signature(depth=1)
print(sig)
```
This code currently segfaults, maybe because the size of each row is larger than expected (width 2, depth 2 would have 3 elements 1, 2, [1,2]).

**Comment by lingyiyang** [comment]
Oh I did not notice a new version came out earlier this week! Yes you are right, the key issue relating to the arrays been viewed in the wrong way is resolved in 0.1.1. 

The above also segfaults for me too, but if reduced to (2 by 3) array, then there is no problem. Perhaps a check should be done first to check compatibility of array size?

The other remark relating to coeffs being displayed as e.g. 1/2 vs 0.5 persists.

**Comment by inakleinbottle** [comment]
OK good. I've checked the cause of the error, it is indeed that the array is larger than expected. This should be a relatively simple fix to take the min of the number of columns and the Lie size of the specified degree.

The coefficient issue is because the context has priority over the the other arguments - if a context is given then the others are simply ignored. Making the context visible at the Python layer was a later decision, and I envisioned that people would primarily create the context and only provide this. Note the default coefficient type is float, so `get_context(width=2, depth=2)` gives a float context, hence the coefficients.
